docs/website/docs/general-usage/credentials/configuration.md

```diff
@@ -41,95 +33,123 @@ def google_sheets(
     credentials=dlt.secrets.value,
     only_strings=False
 ):
-    sheets = build('sheets', 'v4', credentials=Services.from_json(credentials))
+    # allow bot dictionary and a string passed as credentials
+    if isinstance(credentials, str):
+        credentials = json.loads(credentials)
+    # allow both list and comma delimited string to be passed as tabs
+    if isinstance(tab_names, str):
+      tab_names = tab_names.split(",")
+    sheets = build('sheets', 'v4', credentials=ServiceAccountCredentials.from_service_account_info(credentials))
     tabs = []
     for tab_name in tab_names:
-        data = sheets.get(spreadsheet_id, tab_name).execute().values()
+        data = get_sheet(sheets, spreadsheet_id, tab_name)
         tabs.append(dlt.resource(data, name=tab_name))
     return tabs
 ```
+`dlt.source` decorator makes all arguments in `google_sheets` function signature configurable.
+`dlt.secrets.value` and `dlt.config.value` are special argument defaults that tell `dlt` that this
+argument is required and must be passed explicitly or must exist in the configuration. Additionally
+`dlt.secrets.value` tells `dlt` that an argument is a secret.
 
-- `spreadsheet_id`: The unique identifier of the Google Sheets document.
-- `tab_names`: A list of tab names to read from the spreadsheet.
-- `credentials`: Google Sheets credentials as a dictionary ({"private_key": ...}).
-- `only_strings`: Flag to specify if only string data should be retrieved.
-
-`spreadsheet_id` and `tab_names` are configuration values that can be provided directly
-when calling the function. `credentials` is a sensitive piece of information.
-
-`dlt.secrets.value` and `dlt.config.value` are instances of classes that provide
-dictionary-like access to configuration values and secrets, respectively.
-These objects allow for convenient retrieval and modification of configuration
-values and secrets used by the application.
-
-Below, we will demonstrate the correct and wrong approaches to providing these values.
-
-### Wrong approach
-The wrong approach includes providing secret values directly in the code,
-which is not recommended for security reasons.
-
-```python
-# WRONG!:
-# provide all values directly - wrong but possible.
-# secret values should never be present in the code!
-data_source = google_sheets(
-    "23029402349032049",
-    ["tab1", "tab2"],
-    credentials={"private_key": ""}
-)
-```
-:::caution
-Be careful not to put your credentials directly in the code.
+In the example above:
+- `spreadsheet_id`: is a **required config** argument.
+- `tab_names`: is a **required config** argument.
+- `credentials`: is a **required secret** argument (Google Sheets credentials as a dictionary ({"private_key": ...}))
+- `only_strings`: is an **optional config** argument with a default. May be present when calling `google_sheets` function or in the configuration.
```

Comment:
```suggestion
- `only_strings`: is an **optional config** argument with a default value. It may be specified when calling the `google_sheets` function or included in the configuration settings.
```